src/main/java/com/marginallyclever/util/FindAllTraductionGet.java

```diff
@@ -0,0 +1,278 @@
+		if (!n.isEmpty()) {
+		    for (MatchResult mr : n) {
+			countMatchNotLineByLine++;
+			// Can we get the line num ? currently in this implementation we have the car pos in the file/stream ...
+			//	logger.debug(String.format(" %-50s in %s at sart:%d end:%d", mr.group(1), mr.group(), mr.start(), mr.end()));
+		    }
+		}
+	    }
+
+	    // line by line ( this can miss some like "Traduction.\nget(\n...\n);" ) but it gave me the line number in the file/stream ...
+	    Scanner scanner = new Scanner(x);
+	    int lineNum = 0;
+	    while (scanner.hasNextLine()) {
+		lineNum++;
+		String nextToken = scanner.nextLine();
+		Matcher m = pattern.matcher(nextToken);
+		while (m.find()) {
+		    if (debugSearchInAFile) {
+			logger.debug(String.format(" match (groupCount=%d) : \"%s\" line:%d char:%d", m.groupCount(), m.group(0), lineNum, m.start(0)));
+			logger.debug(String.format(" match group(1) : \"%s\" char:%d", m.group(1), m.start(1)));
+			//logger.debug(String.format("match group(1) : \"%s\" in \"%s\" line:%d char:%d", m.group(1), baseDir.relativize(x), lineNum, m.start(1)));
+		    }
+		    if (m.groupCount() >= 1) {
+			FindAllTraductionResult res = new FindAllTraductionResult(m.group(1), lineNum, m.start(1), x);
+			mapMatchResultToFilePath.put(res, x);
+		    } else {
+			// no group found ? bad regexp pattern with no group ? 
+		    }
+		}
+	    }
+	    // warm if we miss some match by doing line by line ?
+	    if (countMatchNotLineByLine != mapMatchResultToFilePath.size()){
+		logger.error(String.format("Miss %d match in multilines ... ",countMatchNotLineByLine-mapMatchResultToFilePath.size()));
+	    }
+	} catch (IOException ex) {
+	    logger.error("{}",ex.getMessage(), ex);
+	}
+	return mapMatchResultToFilePath;
+    }
+
+    /**
+     * List all files and sub files in this path. Using
+     * <code>Files.walk(path)</code> (so this take care of recursive path
+     * exploration ) And applying filter ( RegularFile and ReadableFile ) and
+     * filtering FileName ...
+     *
+     * @param path where to look.
+     * @param fileNameEndsWithSuffix use ".java" to get only ... ( this is not a
+     * regexp so no '.' despecialization required ) can be set to
+     * <code>""</code> to get all files.
+     * @return a list of files (may be empty if nothing is found) or null if
+     * something is wrong.
+     * @throws IOException
+     */
+    public static List<Path> listFiles(Path path, String fileNameEndsWithSuffix) throws IOException {
+	List<Path> result = null;
+	try ( Stream<Path> walk = Files.walk(path)) {
+	    result = walk
+		    .filter(Files::isRegularFile)
+		    .filter(Files::isReadable)
+		    .map(Path::toFile)
+		    .filter(f -> f.getName().endsWith(fileNameEndsWithSuffix))
+		    .map(File::toPath)
+		    .collect(Collectors.toList());
+	}
+	if (debugListFiles) {
+	    if (result == null) {
+		logger.debug(String.format("listFiles(\"%s\",\"%s\")=null", path, fileNameEndsWithSuffix));
```

Comment:
the `logger` API supports format Strings. You can convert this line with:
```
logger.debug("listFiles(\"{}\",\"{}\")=null", path, fileNameEndsWithSuffix);
```

Reply:
Thank you !
And sorry, I should have done it from the start.